.github/workflows/check-sdk-compat.yml

```diff
@@ -0,0 +1,84 @@
+name: Check sdk-go compatibility
+
+on:
+  workflow_dispatch:
+    inputs:
+      sdk_ref:
+        description: sdk-go ref to check ("latest" for latest release tag)
+        required: true
+        default: latest
+      api_ref:
+        description: api-go ref to check
+        required: true
+        default: master
+  workflow_call:
+    inputs:
+      sdk_ref:
+        description: sdk-go ref to check ("latest" for latest release tag)
+        required: true
+        default: latest
+      api_ref:
+        description: api-go ref to check
+        required: true
+        default: master
+
+jobs:
+  check-sdk-go-compatibility:
+    name: "Check sdk-go compatibility"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Validate inputs
+        id: inputs
+        env:
+          GH_TOKEN: ${{ github.token }}
+          SDK_REF: ${{ github.event.inputs.sdk_ref }}
+          API_REF: ${{ github.event.inputs.api_ref }}
+        run: |
+          if [[ "$SDK_REF" == "latest" ]]; then
+            SDK_REF=$(gh api /repos/temporalio/sdk-go/releases/latest --jq '.name')
```

Comment:
I see you're using the `GH_TOKEN` env var so probably not much of a concern but this command may get rate limited by GH.

I vote for removing this step to have one less moving part.

Reply:
The rate limit is 15000 requests per hour: https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-github_token-in-github-actions